tests/functional/parse/dirgraph.test.ts

```diff
@@ -0,0 +1,53 @@
+import { test } from 'tap';
+import { parseDigraphs } from '../../../lib/parse/digraph';
+
+const core = `digraph "io.snyk:core:jar:1.0.0" { 
+"io.snyk:core:jar:1.0.0" -> "org.apache.logging.log4j:log4j-api:jar:2.17.2:compile" ; 
+"io.snyk:core:jar:1.0.0" -> "org.apache.logging.log4j:log4j-core:jar:2.17.2:compile" ; 
+}`;
+
+test('parses Maven CLI digraph to expected MavenGraph', async (t) => {
+  t.same(
+    parseDigraphs([core]),
+    [
+      {
+        rootId: 'io.snyk:core:jar:1.0.0',
+        nodes: {
+          'io.snyk:core:jar:1.0.0': {
+            dependsOn: [
+              'org.apache.logging.log4j:log4j-api:jar:2.17.2:compile',
+              'org.apache.logging.log4j:log4j-core:jar:2.17.2:compile',
+            ],
+          },
+          'org.apache.logging.log4j:log4j-api:jar:2.17.2:compile': {
+            dependsOn: [],
+          },
+          'org.apache.logging.log4j:log4j-core:jar:2.17.2:compile': {
+            dependsOn: [],
+          },
+        },
+      },
+    ],
+    'contains expected graph',
+  );
+});
+
+test('defaults to empty graph with bad input', async (t) => {
+  t.same(
+    parseDigraphs([
+      `bad
+test`,
+    ]),
+    [
+      {
+        rootId: 'root',
+        nodes: {
+          root: {
+            dependsOn: [],
+          },
+        },
+      },
+    ],
+    'returns empty graph',
```

Comment:
If there is no digraph why do we want to return an empty maven graph? Not an error?

Reply:
yea I think that maybe better in hindsight, will make it throw an exception instead, thanks

Reply:
ok pushed a change for that. I remember now, I was half thinking we should fail gracefully and continue to show results for graphs we could parse, but in reality if one parse fails in all likelihood something bad has happened and all output would fail to be parsed